src/System.Net.Sockets/tests/FunctionalTests/SendReceive.cs

```diff
@@ -1133,6 +1133,75 @@ public void SocketSendWouldBlock_ReturnsBytesSent()
                 }
             }
         }
+
+        [Fact]
+        [PlatformSpecific(TestPlatforms.AnyUnix)]
+        [ActiveIssue(31766, TestPlatforms.OSX)]
+        public async Task Socket_ReceiveFlags_Success()
+        {
+            using (var sender = new Socket(AddressFamily.InterNetwork, SocketType.Dgram, ProtocolType.Udp))
+            using (var receiver = new Socket(AddressFamily.InterNetwork, SocketType.Dgram, ProtocolType.Udp))
+            {
+                receiver.BindToAnonymousPort(IPAddress.Loopback);
+                receiver.ReceiveTimeout = TestSettings.PassingTestTimeout;
```

Comment:
Doesn't this only apply to synchronously operations, which this test isn't using?